app.js

```diff
@@ -8,8 +12,20 @@ app.get("/api/topics", getTopics);
 
 app.get("/api", getEndPoints);
 
+app.get("/api/articles/:article_id", getArticleById);
+
 app.all("*", (req, res) => {
   res.status(404).send({ msg: "Endpoint not found!" });
 });
 
+app.use((err, req, res, next) => {
+  if (err.status === 404) {
+    res.status(404).send(err);
+  }
+  if (err.code === "22P02") {
+    res
+      .status(400)
+      .send({ msg: "Bad Request: invalid id (must be an integer)" });
```

Comment:
This is all fine at the moment, but might want to refactor into separate error handlers later on to make it a bit more readable when we've got multiple errors being caught within here 👍